src/main/java/com/limechain/runtime/builder/RuntimeBuilder.java

```diff
@@ -0,0 +1,77 @@
+package com.limechain.runtime.builder;
+
+import com.limechain.config.HostConfig;
+import com.limechain.network.Network;
+import com.limechain.network.protocol.blockannounce.NodeRole;
+import com.limechain.runtime.Runtime;
+import com.limechain.runtime.RuntimeFactory;
+import com.limechain.runtime.hostapi.dto.OffchainNetworkState;
+import com.limechain.storage.KVRepository;
+import com.limechain.storage.crypto.KeyStore;
+import com.limechain.storage.offchain.OffchainStorages;
+import com.limechain.storage.offchain.OffchainStore;
+import com.limechain.storage.offchain.StorageKind;
+import com.limechain.trie.BlockTrieAccessor;
+import io.libp2p.core.Host;
+import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Component;
+
+/**
+ * Serves as a convenience Spring bean to package all Spring beans necessary
+ * for common use-cases regarding building runtime instances.
+ * @implNote Composes over {@link RuntimeFactory}, which is the lower-level granular API for runtime instantiation.
+ */
+@Component
+@RequiredArgsConstructor
+public class RuntimeBuilder {
```

Comment:
Any suggestions for a better name, given the purpose of this class?

Reply:
`RuntimeConfig` || `RuntimeConfigurer` considering it holds beans needed for the base configuration of the runtime.

Reply:
After a brief discussion, this remains as-is for now. Think of it as a known imperfection 😄